overlord/devicestate/devicestate_test.go

```diff
@@ -1315,6 +1317,147 @@ func (s *deviceMgrSuite) TestHasFdeSetupHook(c *C) {
 	}
 }
 
+func (s *deviceMgrSuite) TestRunFdeSetupHookOpInitialSetup(c *C) {
+	st := s.state
+
+	st.Lock()
+	makeInstalledMockKernelSnap(c, st, kernelYamlWithFdeSetup)
+	mockModel := s.makeModelAssertionInState(c, "canonical", "pc", map[string]interface{}{
+		"architecture": "amd64",
+		"kernel":       "pc-kernel",
+		"gadget":       "pc",
+	})
+	devicestatetest.SetDevice(s.state, &auth.DeviceState{
+		Brand: "canonical",
+		Model: "pc",
+	})
+	st.Unlock()
+
+	var hookCalled []string
+	hookInvoke := func(ctx *hookstate.Context, tomb *tomb.Tomb) ([]byte, error) {
+		ctx.Lock()
+		defer ctx.Unlock()
+
+		// check that the context has the right data
+		c.Check(ctx.HookName(), Equals, "fde-setup")
+		var fdeSetup fde.SetupRequest
+		ctx.Get("fde-setup-request", &fdeSetup)
+		c.Check(fdeSetup.Op, Equals, "initial-setup")
+		c.Check(fdeSetup.KeyName, Equals, "some-key-name")
+		c.Check(fdeSetup.Model["series"], DeepEquals, "16")
+		c.Check(fdeSetup.Model["brand-id"], DeepEquals, "canonical")
+		c.Check(fdeSetup.Model["model"], DeepEquals, "pc")
+		c.Check(fdeSetup.Model["grade"], DeepEquals, "unset")
+		c.Check(fdeSetup.Model["signkey-id"], DeepEquals, mockModel.SignKeyID())
+		c.Assert(fdeSetup.Model, HasLen, 5)
```

Comment:
I have a _slight_ preference for complex objects like this to be compared with DeepEquals, since for example we could have a bug where some newly added key is set on the object incorrectly and we forget to update this test and thus miss that we are setting the new thing incorrectly.

Reply:
Yeah, that makes sense, thank you! Updated.